app/controllers/items_controller.rb

```diff
@@ -1,19 +1,39 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_wishlist_and_item, only: %i(edit update)
+  before_action :set_wishlist, only: %i(new create edit update)
+  before_action :set_item, only: %i(edit update)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
```

Comment:
I'm still wondering - maybe we should distinguish those two options. When someone tries to create/edit someone else's wishlist is, redirected to this specific wishlist show page (page is in fact re-rendered). Maybe it's better then being redirected to root which is default?